File: tests/src/joda-time/joda-time/2.9.9/src/test/java/joda_time/joda_time/ZoneInfoProviderAnonymous1Test.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package joda_time.joda_time;

import static org.assertj.core.api.Assertions.assertThat;

import org.joda.time.tz.ZoneInfoProvider;
import org.junit.jupiter.api.Test;

public class ZoneInfoProviderAnonymous1Test {

    @Test
    void loadsZoneInfoMapThroughExplicitClassLoader() throws Exception {
        ZoneInfoProvider provider = new ZoneInfoProvider(
                "org/joda/time/tz/data",
                ZoneInfoProviderAnonymous1Test.class.getClassLoader()
        );

        assertThat(provider.getAvailableIDs()).contains("UTC", "Europe/Paris");
    }

    @Test
    void loadsZoneInfoMapThroughSystemClassLoader() throws Exception {
        ZoneInfoProvider provider = new ZoneInfoProvider("org/joda/time/tz/data", null);

        assertThat(provider.getAvailableIDs()).contains("UTC", "Europe/Paris");
    }
}

File: tests/src/org.playframework.anorm/anorm_3/2.8.1/src/test/scala/org_playframework_anorm/anorm_3/ZoneInfoProviderAnonymous1Test.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_playframework_anorm.anorm_3

import org.joda.time.DateTimeZone
import org.joda.time.tz.ZoneInfoProvider
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class ZoneInfoProviderAnonymous1Test {
  @Test
  def nullLoaderUsesSystemResourceLookup(): Unit = {
    val provider: ZoneInfoProvider = new ZoneInfoProvider("org/joda/time/tz/data", null)

    assertTrue(provider.getAvailableIDs.contains("UTC"))
    assertSame(DateTimeZone.UTC, provider.getZone("UTC"))
    assertNotNull(provider.getZone("Europe/Paris"))
  }
}
